src/System.Runtime.CompilerServices.Unsafe/tests/UnsafeTests.cs

```diff
@@ -432,6 +432,15 @@ public static unsafe void AsRef()
         }
 
         [Fact]
+        public static unsafe void InAsRef()
+        {
+            int[] a = new int[] { 0x123, 0x234, 0x345, 0x456 };
+
+            ref int r = ref Unsafe.AsRef<int>(a[0]);
+            Assert.Equal(0x123, r);
```

Comment:
It maybe also useful to try to assign to `r` and verify that the value shows up in `a[0]`.

Reply:
Isn't that only guaranteed if the `in` keyword is specified at the call-site? (The current Roslyn version does not seem to support that yet.)

Reply:
The `in` syntax for readonly refs was removed (https://github.com/dotnet/roslyn/pull/22182) because of it was redundant. I do not think it is coming back.

Reply:
This was changed to be `in` rather than `ref readonly` approx 10 days after 22182: https://github.com/dotnet/roslyn/issues/22387

Reply:
‘in’ provides guarantees of aliasing. No modifier at callsite just makes the best effort. In simple cases there is no difference. Complex cases are not interesting here since we are not testing the compiler.  

It would be interesting though to get a ref to something readonly - like a readonly field, assign a new value, and observe the changed value by directly reading the field (not via AsRef).

Reply:
@ektrah Do you plan to make this addition to the test? E.g. something like:

```
r = 0x42;
Assert.Equal(0x42, a[0]);
```

Reply:
@jkotas Given that "No modifier at callsite just makes the best effort." it seems that such as test could randomly fail with the current compiler version. So I would keep the test as it is right now. I'm happy to provide an update later when a syntax to guarantee aliasing is available.

Reply:
It cannot fail randomly. The compiler has to be deterministic.

Reply:
The behavior is not random :). Lvalues of compatible type will not be passed via copy.

It would be a good practice to use “in” at call site for methods like this though - since accidentally getting in situation when copy is made would be undesirable.

Reply:
OK, I'm convinced. I've added the above check to the test.